Make boinc compile with LDFLAGS="-Wl,--as-needed"

2009-09-28 Martin von Gagern

References:
http://bugs.gentoo.org/286771
http://www.gentoo.org/proj/en/qa/asneeded.xml

Index: boinc-6.6.40/lib/Makefile.am
===================================================================
--- boinc-6.6.40.orig/lib/Makefile.am
+++ boinc-6.6.40/lib/Makefile.am
@@ -103,7 +103,7 @@ libboinc_la_SOURCES = $(generic_sources)
 libboinc_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_LDFLAGS = -L$(libdir) -rpath $(libdir) -version-number $(LIBBOINC_VERSION)
-libboinc_la_LIBADD =
+libboinc_la_LIBADD = $(SSL_LIBS)
 
 if ENABLE_FCGI
 lib_LTLIBRARIES += libboinc_fcgi.la
